ocaml/xapi/message_forwarding.ml

```diff
@@ -922,8 +922,12 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
 						(fun () ->
 							finally_clear_host_operation ~__context ~host:suitable_host ?host_op ();
 							(* In certain cases, VM might have been destroyed as a consequence of operation *)
-							if Db.is_valid_ref __context vm then
-								Db.VM.set_scheduled_to_be_resident_on ~__context ~self:vm ~value:Ref.null))
+							if Db.is_valid_ref __context vm then begin
+								Db.VM.set_resident_on ~__context ~self:vm ~value:Ref.null;
+								List.iter
+									(fun vgpu -> Db.VGPU.set_scheduled_to_be_resident_on ~__context ~self:vgpu ~value:Ref.null)
```

Comment:
This isn't relevant in cream and makes some of the future backported patches misleading.